### PR TITLE
[chore][pkg/stanza] Add syslog octet counting test cases with whitespace scenarios

### LINKED ISSUE #31477: [receiver/syslog] incorrect parsing if message ends in a space
### Component(s)

receiver/syslog

## Description
Using RFC5424 format with octet framing.  If the message ends in a space, otel logs an error and fails to parse the log attributes.

## Steps to Reproduce
```
echo -n "84 <13>1 2024-02-28T03:32:00.313226+00:00 192.168.65.1 inactive - - -  partition is p2 " | nc localhost 40514 -w 0
```

## Expected Result
The message should be parsed with all attributes intact.  If I remove the space and reduce the MSG-LEN by 1 the result is correct.
```
echo -n "83 <13>1 2024-02-28T03:32:00.313226+00:00 192.168.65.1 inactive - - -  partition is p2" | nc localhost 40514 -w 0
```
Correct debug output looks like this:
```
LogRecord #0
ObservedTimestamp: 2024-02-28 08:51:37.525827053 +0000 UTC
Timestamp: 2024-02-28 03:32:00.313226 +0000 UTC
SeverityText: notice
SeverityNumber: Info2(10)
Body: Str(83 <13>1 2024-02-28T03:32:00.313226+00:00 192.168.65.1 inactive - - -  partition is p2)
Attributes:
     -> hostname: Str(192.168.65.1)
     -> appname: Str(inactive)
     -> version: Int(1)
     -> priority: Int(13)
     -> message: Str( partition is p2)
     -> facility: Int(1)
```

## Actual Result
```
2024-02-28T08:53:27.143Z	error	helper/transformer.go:98	Failed to process entry	{"kind": "receiver", "name": "syslog/local", "data_type": "logs", "operator_id": "syslog_input_internal_parser", "operator_type": "syslog_parser", "error": "found EOF after \"<13>1 2024-02-28T03:32:00.313226+00:00 192.168.65.1 inactive - - -  partition is p2\", expecting a SYSLOGMSG containing 84 octets", "action": "send"}
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*TransformerOperator).HandleEntryError
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.95.0/operator/helper/transformer.go:98
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*ParserOperator).ParseWith
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.95.0/operator/helper/parser.go:140
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*ParserOperator).ProcessWithCallback
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.95.0/operator/helper/parser.go:112
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/parser/syslog.(*Parser).Process
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.95.0/operator/parser/syslog/syslog.go:183
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*WriterOperator).Write
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.95.0/operator/helper/writer.go:53
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/tcp.(*Input).handleMessage
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.95.0/operator/input/tcp/tcp.go:321
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/tcp.(*Input).goHandleMessages.func1
	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.95.0/operator/input/tcp/tcp.go:282
2024-02-28T08:53:27.155Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 1}
2024-02-28T08:53:27.155Z	info	ResourceLog #0
Resource SchemaURL: 
ScopeLogs #0
ScopeLogs SchemaURL: 
InstrumentationScope  
LogRecord #0
ObservedTimestamp: 2024-02-28 08:53:27.143189548 +0000 UTC
Timestamp: 1970-01-01 00:00:00 +0000 UTC
SeverityText: 
SeverityNumber: Unspecified(0)
Body: Str(84 <13>1 2024-02-28T03:32:00.313226+00:00 192.168.65.1 inactive - - -  partition is p2)
```

### Collector version

v0.95.0

## Environment
OS: Ubuntu 22.04

### OpenTelemetry Collector configuration

```yaml
receivers:
  syslog/local:
    tcp:
      listen_address: "127.0.0.1:40514"
    protocol: rfc5424
    enable_octet_counting: true
exporters:
  debug:
    verbosity: detailed
    sampling_initial: 5
    sampling_thereafter: 200
service:
  pipelines:
    logs:
      receivers:
        - syslog/local
      exporters:
        - debug
```

### Log output

_No response_

### Additional context

I used netcat to reproduce this simply, but in reality I'm using rsyslog to receive messages and forward to otel.  The same issue happens there, and rsyslog DebugFormat output looks correct.

### PR DESCRIPTION
**Pull request by djaglowski**
Resolves #31477

### CODE REVIEW

File: receiver/syslogreceiver/README.md
Comment (tiffany76) [inline receiver/syslogreceiver/README.md:87]:
```suggestion
**note** If `multiline` is not set at all, it won't split log entries at all. Every UDP packet is going to be treated as log.
```

File: receiver/syslogreceiver/README.md
Comment (tiffany76) [inline receiver/syslogreceiver/README.md:102]:
```suggestion
| `utf-16be` | UTF-16 encoding with big-endian byte order                    |
```

**Review by tiffany76 — COMMENTED** [review]
Docs reviewer here. Just a couple typo corrections. Thanks!

**Comment by djaglowski** [comment]
This is just waiting on a review @open-telemetry/collector-contrib-approvers

File: receiver/syslogreceiver/README.md
Comment (crobert-1) [inline receiver/syslogreceiver/README.md:87]:
```suggestion
**note** If `multiline` is not set at all, it won't split log entries at all. Every UDP packet is going to be treated as a log.
```

File: receiver/syslogreceiver/README.md
Comment (crobert-1) [inline receiver/syslogreceiver/README.md:112]:
```suggestion
If set, the `async` configuration block instructs the `udp_input` operator to read and process logs asynchronously and concurrently.
```

File: pkg/stanza/operator/parser/syslog/data.go
Comment (djaglowski) [reply pkg/stanza/operator/parser/syslog/data.go:344]: Yeah, it's hacky but I don't see a better way that doesn't require a ton of refactoring. A comment is a good idea.

File: pkg/stanza/operator/input/syslog/input_test.go
Comment (djaglowski) [reply pkg/stanza/operator/input/syslog/input_test.go:121]: It actually does. The flag is set just below, and the extra whitespace is subtle but included at the beginning of the syslog message.

Comment (crobert-1) [reply pkg/stanza/operator/input/syslog/input_test.go:121]: I'm pretty ignorant with syslog formats, so just to confirm, the actual syslog message in this case is ` partition is p2 `, with a leading and tailing whitespace character?

Comment (djaglowski) [reply pkg/stanza/operator/input/syslog/input_test.go:121]: More or less. To slightly rephrase, there is a "message" within the syslog which has a leading and trailing space that is preserved.

**Review by crobert-1 — APPROVED** [review]

**Pull request merged**
